Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -310,14 +310,14 @@ - (void)transactionWithRetries:(int)retries
           if (!maybe_result.ok()) {
             if (retries > 0 && [self isRetryableTransactionError:maybe_result.status()] &&
                 !transaction->IsPermanentlyFailed()) {
-              workerQueue->VerifyIsCurrentQueue();
               return [self transactionWithRetries:(retries - 1)
                                       workerQueue:workerQueue
                                    updateCallback:updateCallback
                                    resultCallback:resultCallback];
+            } else {
+              resultCallback(std::move(maybe_result));
+              return;
```

Comment:
Er, sorry...  Looking closer, why do we even have these returns?  Doesn't this function return void?  I think we can drop them which makes the code even clearer (I think).

Separately, please make a similar change to the below retries block of code too.